### PR TITLE
fix: nil pointer

### LINKED ISSUE #4186: [BUG] func GetSubnetIPRangeString nil pointer dereference
### Kube-OVN Version

v1.13.0

### Kubernetes Version

v1.25.3

### Operation-system/Kernel Version

"Ubuntu 22.04.4 LTS" 6.5.0-35-generic

### Description

image

### Steps To Reproduce

1、手动指定Sunbet的spec.excludeIps字段，指定一个子网外的IP，或者类似这种192.168.0.10..192.168.0.9前面IP大，后面小的情况
2、此时kube-ovn-controller会抛出异常，但不会出现空指针
3、重启kube-ovn-controller，在init.go中会触发空指针异常导致kube-ovn-controller退出

### Current Behavior

重启kube-ovn-controller后pod起不来

### Expected Behavior

重启kube-ovn-controller后pod能正常运行

### PR DESCRIPTION
**Pull request by zcq98**
# Pull Request

- [x] Make sure you have followed [Kube-OVN Code Style](https://github.com/kubeovn/kube-ovn/blob/master/CODE_STYLE.md).

## What type of this PR

Examples of user facing changes:

- Bug fixes

## Which issue(s) this PR fixes

Fixes #4186

### CODE REVIEW

**Comment by oilbeater** [comment]
@zcq98 can you add a ut for this?

**Comment by zhangzujian** [comment]
在 IPAM Init 过程中判断 subnet 是否 ready，是否更合理？

**Comment by zcq98** [comment]
> 在 IPAM Init 过程中判断 subnet 是否 ready，是否更合理？

好的，我再去看看代码

File: pkg/controller/init.go [hunk 1]
```diff
@@ -643,6 +643,17 @@ func (c *Controller) syncSubnetCR() error {
 	}
 	for _, cachedSubnet := range subnets {
 		subnet := cachedSubnet.DeepCopy()
+		isReady := true
+		for _, condition := range subnet.Status.Conditions {
+			if (condition.Type == kubeovnv1.Ready || condition.Type == kubeovnv1.Validated) && condition.Status != v1.ConditionTrue {
```
Comment (zhangzujian) [inline pkg/controller/init.go:648]: Use `subnet.Status.IsReady()`.

**Review by zhangzujian — APPROVED** [review]

**Pull request merged**